app/carbonapi/http_handlers.go

```diff
@@ -802,7 +805,7 @@ func (app *App) infoHandler(w http.ResponseWriter, r *http.Request) {
 
 	logAsError := false
 	defer func() {
-		deferredAccessLogging(r, &accessLogDetails, t0, logAsError)
+		app.deferredAccessLogging(r, &accessLogDetails, t0, logAsError)
```

Comment:
What took my attention is: this defer call is spread all around the code, even though it's always being called with the same set of arguments. I'm curious if this can be put as a single statement in the very end of the method call and, maybe, this can un-defer it entirely.

Reply:
You are probably right. This code needs a big refactoring, and this will be part of it.